### ISSUE TITLE
Character 'Y' switching between two keyboard layouts (cz -> rus)

### ISSUE THREAD

**Issue by rampaq** [jordansissel/xdotool#354]
There is also a weird issue with _two keyboard layouts_ in _3.20210804.1/2_, which was not present in previous versions.

I set a primary Czech (`cz`) keyboard and add a russian phonetic layout, toggled between by Alt+Shift.
`setxkbmap -layout cz,cz -variant ,rus -option grp:alt_shift_toggle`

Then a capital 'Y' types as russian 'Ы' for some reason.
```
echo "Y" | xdotool type --clearmodifiers --file -                                             
# output: Ы

echo "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz" | xdotool type --clearmodifiers --file -                                             
# output: ABCDEFGHIJKLMNOPQRSTUVWXЫZabcdefghijklmnopqrstuvwxyz
```

What is interesting that this happens only with capital 'Y'. I thought that xdotool pushes the _toggle_ switch command but that does not seem to be the case as `setxkbmap -layout cz,cz -variant ,rus -option grp:alt_space_toggle` (changed toggle key) behaves exactly the same.

_However_, when I change the primary layout to `us` then everything is ok:
```
setxkbmap -layout us,us -variant ,rus -option grp:alt_space_toggle
echo "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz" | xdotool type --clearmodifiers --file -
ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz
```

And yes, the 'Y' is still ascii hex 59 on both keyboards.

Note: `setxkbmap` after setting the layout does not solve the problem, as was suggested by some in #97

**Comment by jordansissel** [comment]
Weird key map problems in xdotool are An old but unsolved issue. Thank you for the great details in your report! 

I don’t have any suggestions right now, but I am hopeful we can find a fix soon.

**Comment by rampaq** [comment]
Yes, but this is only a matter of the latest version, as I did not noticed any problems before.

In fact, it is caused by a supposed fix for Latin letters (PR #283). I built `xdotool` without that PR and it works! Quite baffled why does setting Shift modifier causes this, also only for this one 'Y' letter. 

How does xdotool handle unicode characters btw?

Y = 59, Ы = (unicode) d0 ab 
all hex, from `hexdump`

**Comment by jordansissel** [comment]
Oh that’s good to know — I’ll check on your test case (Y vs Ы, with and
without #283) and see what I can find.

I appreciate you testing the code in different ways to find a likely cause
of the bug :)

**Comment by jordansissel** [comment]
I tested this a bit and found that with your keymap, the 'Y' key is available two different ways. The first way is what xdotool chooses and it doesn't type it correctly (as you noted).

xdotool sends the "Shift" key then the Cyrillic_yeru `ы` key, when it should be sending a different sequence of modifiers (shift, alt, etc). I'll need to dig more to figure out what xdotool is doing wrong.

**Comment by rampaq** [comment]
Thanks for pursuing this further! If you need any testing, please let me know (I'm using `3.20160805.1-3` for now, haven't tested newer releases).
